digital_twin_advanced.py

```
@@ -0,0 +1,453 @@
+            else:
+                response = self._mock_response(user_message, relevant_memories)
+            
+            return response
+        
+        except Exception as e:
+            print(f"\n⚠️  LLM error: {e}")
+            return self._mock_response(user_message, relevant_memories)
+    
+    def _call_openai(self, system_prompt: str, messages: List[Dict]) -> str:
+        """Roep OpenAI API aan"""
+        full_messages = [{"role": "system", "content": system_prompt}] + messages
```

Ruff (0.14.10)

Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)